src/main/java/org/opensearch/security/privileges/PitPrivilegesEvaluator.java

```diff
@@ -0,0 +1,193 @@
+            }
+        } catch(InterruptedException e) {
+            Thread.currentThread().interrupt();
+            log.error(e.toString());
+        }
+        return presponse;
+    }
+
+    /**
+     * Handle access for Get All PITs access
+     */
+    private PrivilegesEvaluatorResponse handleGetAllPitsAccess(final ActionRequest request, final ClusterService clusterService,
+                                                               final User user, SecurityRoles securityRoles, final String action,
+                                                               IndexNameExpressionResolver resolver,
+                                                               boolean dnfOfEmptyResultsEnabled, PrivilegesEvaluatorResponse presponse) throws InterruptedException {
+        List<ListPitInfo> pitInfos = ((GetAllPitNodesRequest) request).getGetAllPitNodesResponse().getPitInfos();
+        // if cluster has no PITs, then allow the operation to pass with empty response if dnfOfEmptyResultsEnabled
+        // config property is true, otherwise fail the operation
+        if(pitInfos.isEmpty()) {
+            if(dnfOfEmptyResultsEnabled) {
+                presponse.allowed = true;
+                presponse.markComplete();
+            }
+            return presponse;
+        }
+        List<String> pitIds = new ArrayList<>();
+        pitIds.addAll(pitInfos.stream().map(ListPitInfo::getPitId).collect(Collectors.toList()));
+        Map<String, String[]> pitToIndicesMap = OpenSearchSecurityPlugin.GuiceHolder.getPitService().getIndicesForPits(pitIds);
+        Map<String, ListPitInfo> pitToPitInfoMap =  new HashMap<>();
+
+        for(ListPitInfo pitInfo : pitInfos) {
+            pitToPitInfoMap.put(pitInfo.getPitId(), pitInfo);
+        }
+        List<ListPitInfo> permittedPits = new ArrayList<>();
+        for (String pitId : pitIds) {
+            String[] indices = pitToIndicesMap.get(pitId);
+            HashSet<String> indicesSet = new HashSet<>(Arrays.asList(indices));
+
+            final ImmutableSet<String> INDICES_SET = ImmutableSet.copyOf(indicesSet);
+            final IndexResolverReplacer.Resolved pitResolved =
+                    new IndexResolverReplacer.Resolved(INDICES_SET, INDICES_SET, INDICES_SET,
+                            ImmutableSet.of(), SearchRequest.DEFAULT_INDICES_OPTIONS);
+
+            final Set<String> allPermittedIndices = securityRoles.reduce(pitResolved,
+                    user, new String[]{action}, resolver, clusterService);
+            if(isDebugEnabled) {
+                log.debug("Evaluating PIT ID : " + pitId );
+            }
+            if (allPermittedIndices.size() == INDICES_SET.size()) {
+                if(isDebugEnabled) {
+                    log.debug(" Permitting PIT ID : " + pitId);
+                }
+                permittedPits.add(pitToPitInfoMap.get(pitId));
+            }
+        }
+        if (permittedPits.size() > 0) {
```

Comment:
We are making this whole change because we want the pit permission to be granular - only the allowed PITs are shown to the user ( by setting the permitted pits in request )